xrdp/xrdp_wm.c

```diff
@@ -1722,15 +1722,27 @@ get_unicode_character(struct xrdp_wm *self, int device_flags, char16_t c16)
 static int
 xrdp_wm_key_unicode(struct xrdp_wm *self, int device_flags, char32_t c16)
 {
+    int index;
     char32_t c32 = get_unicode_character(self, device_flags, c16);
 
     if (c32 == 0)
     {
         return 0;
     }
 
-    int index;
+    // Send the character to chansrv if it's capable of doing something
+    // with it
+    if (self->mm->chan_trans != NULL &&
+            self->client_info->unicode_input_support == UIS_ACTIVE &&
+            self->mm->chan_trans->status == TRANS_STATUS_UP)
+    {
+        xrdp_mm_send_unicode_to_chansrv(self->mm,
```

Comment:
There is no return status for "xrdp_mm_send_unicode_to_chansrv" to tell unicode processing failure here for the fall back in line 1746. Therefore it's possible that the fall back not working. For example, the unicode character is fail to process in the unicode thread but it does exist in the keymap.

My suggestion is to put the unicode procedure in the end of the function. You know, the result is the same if the character is existing in keymap, but it's more stable in the old way, I think.

Reply:
I'm fine with that. I'll change it back.